atlp-rwanda/atlp-devpulse-bn
# #147 Implementing google authentication

## Description
This facilitates google login by adding various mutations and querries to the backend

### Description of tasks that were expected to be completed

- A mutation that allows a logged user to edit their own account
- Adding the `AuthMethod` field to the user model to allow easy verification of authentication methods
-  Adding a querry that retrieves a user using a certain filter could be email, firstname, lastname, country,...


### How has this been tested?
- Clone repo
- Checkout branch `fix-google-auth`
- Set the environment variables as in  `.env.example`
- Run `npm install`, then `npm run dev` to check if the app is running successfully
- Navigate to `localhost:5000` and query the server

- `mutation UpdateUserSelf` to update your data 
```
{  id,
userInput: {
firstname: "Kalisa"
country: "Congo"
  
}
```
- Use query `getByFilter` to retrieve users with a certain filter

### Track PR (issue number & link)
The issue link is the on front end click [here](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/147)

### Screenshots (check the frontend)
N/A
